### DISCUSSION TITLE
ASA in multiple context mode config file move to new ASA

### DISCUSSION THREAD

**Discussion by gwankius** [ktbyers/netmiko#3293] [Q&A]
Hallo team,

I would like to migrate all the config files off an ASA in multiple context mode to a new ASA.
As there could be i.e 30+ context on the ASA is there a way to automate this or is the best way done with the netmiko scp file transfer.
Is there somebody who has done this before and willing to share some pointers 
As i am new to network automation/netmiko i found the script for file transfer and modified like below..

```python
#!/usr/bin/env python
import os
from getpass import getpass
from netmiko import ConnectHandler, file_transfer

cisco = {
    "device_type": "cisco_asa",
    "host": "x.x.x.x",
    "username": "my_name",
    "password": password,
    "file_system": "disk0:",
    "port": 22
}

# looping over the config files of the different context on my server
for file in source_files:
    source_file = file
    dest_file = file
    direction = put 
	
for net_device in (cisco):
    file_system = net_device.pop("file_system")

    # Create the Netmiko SSH connection
    ssh_conn = ConnectHandler(**net_device)
    transfer_dict = file_transfer(
        ssh_conn,
        source_file=source_file,
        dest_file=dest_file,
        file_system=file_system,
        direction=direction,
        overwrite_file=True,
    )
    print(transfer_dict)
    pause = input("Hit enter to continue: ")
```

**Comment by ktbyers** [comment]
@gwankius If you look at the file system on the ASA where are all the config files stored? Can you see all of the config files from one context?

**Reply by gwankius** [reply]
Hi Kirk, i was thinking to have the config files locally on my server and upload them from there to the new asa, as all config files from the different context are on disk0 on the old asa. So most likely first will be a get from the old asa and put to the new asa

**Reply by ktbyers** [reply]
Yes, but what (roughly) is the path to the config files on the old ASA. Can get all the config files from a single context or do you need to switch contexts to get them?

Only talking about the old ASA here.

**Comment by gwankius** [comment]
Well when logging in to the old asa via ssh will get you to the admin
context and from there you have access to all config files on the asa kept
at disk0. So no need to move to the different contexts



Op ma 18 sep. 2023 16:37 schreef Kirk Byers ***@***.***>:

**Reply by ktbyers** [reply]
Okay, to answer the first part of this, yes, I would probably just do an SCP file transfer on each of those files and then retrieve them and write them out locally (to the system running my Python script).

**Reply by ktbyers** [reply]
Was there an additional part to this question?

**Reply by gwankius** [reply]
Thank you for your reply will build and test next..

**Reply by ktbyers** [reply]
Yeah, let me know if you have questions/run into issues on the code. I would probably make a list of directories/filenames where all the config files lived and then just transfer them one by one in a loop (to a local directory on the computer running the script).

**Comment by gwankius** [comment]
Is it correct to say "changeto context" is not supported in the ssh module?

**Reply by ktbyers** [reply]
No, it should work fine (though you have to get the steps/code correct).

**Comment by gwankius** [comment]
Thanks, it worked. I can see what is missing.. now get the correct output


Op di 26 sep 2023 om 04:14 schreef Kirk Byers ***@***.***>:
